Error Reporting with match_examples broke in 0.7.7

I'm heavily using `exc.match_examples` in radish: https://github.com/radish-bdd/radish/blob/master/src/radish/parser/core.py#L124

Unfortunately, the release 0.7.7 broke all my code regarding that.
I've seen in the release notes, that you've made some improvements there. 
What exactly did you change, so that my examples are not applicable any longer? 
Am I using this wrong? Or did those improvements introduce a bug? 

You can easily reproduce it by executing the parser tests in radish:

```
tox -e py36 -- tests/unit/test_parser.py
```

And also see the passing version with the pre-release build: 0.7.6a1

I'm already checking for `UnexpectedInput`. 

The error from the parser is correctly: 

```
E           lark.exceptions.UnexpectedToken: Unexpected token Token(_NEWLINE, '\n') at line 4, column 18.
E           Expected one of: 
E           	* STRING_NO_NL
```

but then the `match_examples` method complains that the operation is not supported:

```
    def match_examples(self, parse_fn, examples):
        """ Given a parser instance and a dictionary mapping some label with
            some malformed syntax examples, it'll return the label for the
            example that bests matches the current error.
        """
>       assert self.state is not None, "Not supported for this exception"
E       AssertionError: Not supported for this exception
```

It seems to me that the exception has no `state` assigned?

--- a/lark/lexer.py
+++ b/lark/lexer.py
@@ -384,6 +384,6 @@ class ContextualLexer(Lexer):
 
             value, type_ = root_match
             t = Token(type_, value, e.pos_in_stream, e.line, e.column)
-            raise UnexpectedToken(t, e.allowed)
+            raise UnexpectedToken(t, e.allowed, state=e.state)
 
 ###}

